pkg/sitewise/api/list_associated_assets.go

```diff
@@ -9,45 +9,91 @@ import (
 	"github.com/grafana/iot-sitewise-datasource/pkg/framer"
 	"github.com/grafana/iot-sitewise-datasource/pkg/models"
 	"github.com/grafana/iot-sitewise-datasource/pkg/sitewise/client"
-	"github.com/grafana/iot-sitewise-datasource/pkg/util"
 )
 
 func ListAssociatedAssets(ctx context.Context, client client.SitewiseClient, query models.ListAssociatedAssetsQuery) (*framer.AssociatedAssets, error) {
 
 	var (
 		hierarchyId        *string
 		traversalDirection *string
-		assetId            *string = util.GetAssetId(query.BaseQuery)
 		results            []*iotsitewise.AssociatedAssetsSummary
 	)
 
-	// Recursively load children
-	if query.LoadAllChildren {
-		asset, err := client.DescribeAsset(&iotsitewise.DescribeAssetInput{
-			AssetId: assetId,
-		})
+	seenAssetIds := make(map[string]bool)
 
-		if err != nil {
-			return nil, err
-		}
+	for _, assetId := range query.BaseQuery.AssetIds {
+		assetIdPtr := aws.String(assetId)
+
+		// Recursively load children
+		if query.LoadAllChildren {
+			asset, err := client.DescribeAsset(&iotsitewise.DescribeAssetInput{
+				AssetId: assetIdPtr,
+			})
+
+			if err != nil {
+				return nil, err
+			}
+
+			for _, h := range asset.AssetHierarchies {
+				var nextToken *string = nil
+
+				for {
+					resp, err := client.ListAssociatedAssetsWithContext(ctx, &iotsitewise.ListAssociatedAssetsInput{
+						AssetId:     assetIdPtr,
+						HierarchyId: h.Id,
+						MaxResults:  MaxSitewiseResults,
+						NextToken:   nextToken,
```

Comment:
I noticed that the original code use "" for next token for the first case and doesn't return the token and for the other case used `getNextToken(query.BaseQuery)` and did return it. Was the second one a bug?

Reply:
The original code would handle fetching all of the associated assets so it didn't have to return a next token. The other case didn't have that handling before.